server/api/db/sqldb/db.py

```diff
@@ -2692,22 +2692,48 @@ def refresh_project_summaries(
         session: Session,
         project_summaries: list[mlrun.common.schemas.ProjectSummary],
     ):
-        project_summaries_to_update = []
-        for project_summary_schema in project_summaries:
-            # To avoid a race condition with the delete project flow where a project might be deleted
-            # after its summary has been queried but before the transaction is completed, we check
-            # that the project summary is not `None` after querying.
-            # This is why we cannot query all project summaries outside the loop.
-            query = self._query(
-                session, ProjectSummary, project=project_summary_schema.name
+        """
+        This method updates the summaries of projects that have associated projects in the database
+        and removes project summaries that no longer have associated projects.
+        """
+
+        summary_dict = {summary.name: summary.dict() for summary in project_summaries}
```

Comment:
```suggestion
        summary_dicts = {summary.name: summary.dict() for summary in project_summaries}
```

Reply:
Another thing, not super related, but it is a bit confusing that `models.ProjectSummary` has a `project` attribute, but `schemas.ProjectSummary` has a `name` attribute, which both mean the same.
I think we should align them both to `project`

Reply:
> Another thing, not super related, but it is a bit confusing that `models.ProjectSummary` has a `project` attribute, but `schemas.ProjectSummary` has a `name` attribute, which both mean the same.
> I think we should align them both to `project`

@TomerShor I can't do that, it is not BC

Reply:
right, only the model is new...